app/integration_action.go

```diff
@@ -140,6 +127,50 @@ func (a *App) DoPostActionWithCookie(postId, actionId, userId, selectedOption st
 		upstreamURL = action.Integration.URL
 	}
 
+	// get channel, user and team details
+	chanChan := make(chan store.StoreResult, 1)
+	go func() {
+		channel, err := a.Srv.Store.Channel().Get(upstreamRequest.ChannelId, true)
+		chanChan <- store.StoreResult{Data: channel, Err: err}
+		close(chanChan)
+	}()
+
+	userChan := make(chan store.StoreResult, 1)
+	go func() {
+		user, err := a.Srv.Store.User().Get(upstreamRequest.UserId)
+		userChan <- store.StoreResult{Data: user, Err: err}
+		close(userChan)
+	}()
+
+	cr := <-chanChan
+	if cr.Err != nil {
+		return "", cr.Err
+	}
+	channel := cr.Data.(*model.Channel)
+	upstreamRequest.ChannelName = channel.Name
+	upstreamRequest.TeamId = channel.TeamId
+
+	ur := <-userChan
+	if ur.Err != nil {
+		return "", ur.Err
+	}
+	user := ur.Data.(*model.User)
+	upstreamRequest.UserName = user.Username
+
+	teamChan := make(chan store.StoreResult, 1)
```

Comment:
This is another db request we are starting. We can't start it at the beginning of the method because we don't know `channel.TeamId`. But we can start it as soon 'cchan` return. Would you please change it do run as soon as possible?

Reply:
I'll make the change, thanks

Reply:
I might have missed it, but looks already taken care of?

Reply:
Yes, fixed in 30e8548, thanks